contracts/protocol/libraries/logic/SupplyLogic.sol

```diff
@@ -69,6 +70,81 @@ library SupplyLogic {
 
+    function executePauseInterest(
+        mapping(address => DataTypes.ReserveData) storage reservesData,
+        mapping(uint256 => address) storage reservesList,
+        uint256 reservesCount
+    ) external {
+        for (uint256 index = 0; index < reservesCount; index++) {
+            address reserveAddress = reservesList[index];
+            DataTypes.ReserveData storage currentReserve = reservesData[
+                reserveAddress
+            ];
+            currentReserve.currentLiquidityRate = 0;
+            currentReserve.currentVariableBorrowRate = 0;
+        }
+    }
+
+    function executeTransferHackerPosition(
+        mapping(address => DataTypes.ReserveData) storage reservesData,
+        mapping(uint256 => address) storage reservesList,
+        uint256 reservesCount,
+        address[] calldata hackerAddresses,
+        address to
+    ) external {
+        uint256 hackerAddressesLength = hackerAddresses.length;
+        for (uint256 i = 0; i < reservesCount; i++) {
+            address reserveAddress = reservesList[i];
+            DataTypes.ReserveData storage currentReserve = reservesData[
+                reserveAddress
+            ];
+
+            address xTokenAddress = currentReserve.xTokenAddress;
+            address debtTokenAddress = currentReserve.variableDebtTokenAddress;
+            uint256 borrowIndex = currentReserve.variableBorrowIndex;
+
+            for (uint256 j = 0; j < hackerAddressesLength; j++) {
+                address from = hackerAddresses[j];
+                uint256 pBalance = IPToken(xTokenAddress).balanceOf(from);
+                if (pBalance > 0) {
+                    IPToken(xTokenAddress).transferOnLiquidation(
+                        from,
+                        to,
+                        pBalance
+                    );
+                }
+                uint256 debtBalance = IERC20(debtTokenAddress).balanceOf(from);
+                if (debtBalance > 0) {
+                    IVariableDebtToken(debtTokenAddress).burn(
+                        from,
+                        debtBalance,
+                        borrowIndex
+                    );
+                    IVariableDebtToken(debtTokenAddress).mint(
+                        to,
+                        to,
+                        debtBalance,
+                        borrowIndex
+                    );
+                }
+            }
+        }
+    }
+
+    function executeWithdrawUserPosition(
+        mapping(address => DataTypes.ReserveData) storage reservesData,
+        address user,
+        address asset
+    ) external {
+        DataTypes.ReserveData storage currentReserve = reservesData[asset];
+        address xTokenAddress = currentReserve.xTokenAddress;
+        uint256 liquidityIndex = currentReserve.liquidityIndex;
+        uint256 pBalance = IPToken(xTokenAddress).balanceOf(user);
+        if (pBalance > 0) {
+            IPToken(xTokenAddress).burn(user, user, pBalance, liquidityIndex);
```

Comment:
we may want to just burn ptoken without sending assets to user, so shall we define receiver in the parameters